lua/lualine/config.lua

```diff
@@ -6,6 +6,265 @@ local modules = require('lualine_require').lazy_require {
   utils_notices = 'lualine.utils.notices',
 }
 
+---@alias LualineThemes
+---| "auto"
+---| "16color"
+---| "ayu_dark"
+---| "ayu_light"
+---| "ayu_mirage"
+---| "ayu"
+---| "codedark"
+---| "dracula"
+---| "everforest"
+---| "gruvbox_dark"
+---| "gruvbox_light"
+---| "gruvbox"
+---| "gruvbox-material"
+---| "horizon"
+---| "iceberg_dark"
+---| "iceberg_light"
+---| "iceberg"
+---| "jellybeans"
+---| "material"
+---| "modus-vivendi"
+---| "molokai"
+---| "moonfly"
+---| "nightfly"
+---| "nord"
+---| "OceanicNext"
+---| "onedark"
+---| "onelight"
+---| "palenight"
+---| "papercolor_dark"
+---| "papercolor_light"
+---| "PaperColor"
+---| "powerline"
+---| "powerline_dark"
+---| "pywal"
+---| "seoul256"
+---| "solarized_dark"
+---| "solarized_light"
+---| "Tomorrow"
+---| "wombat"
+
+---@class LualineDisabledFiletypes
+---only ignores the ft for statusline
+---@field statusline table
+---only ignores the ft for winbar.
+---@field winbar table
+
+---@class LualineRefresh
+---@field statusline integer
+---@field tabline integer
+---@field winbar integer
+
+---@class LualineOpts
+-- Show icons.
+---@field icons_enabled boolean
+-- Lualine theme name.
+-- Auto theme will load the theme automaically for your colorscheme
+-- or will generate one otherwise.
+---@field theme LualineThemes
+-- Separators between components.
+---@field component_separators table|string
```

Comment:
make the table part of component&section separators a class that have left and right field